### PR TITLE
Type `release/__init__.py`

### PR DESCRIPTION
**Pull request by jeschkies**
## High-level description

This patch introduces type checks for `release/__init__.py`. It should enable us to refactor and split up the release scripts.

### CODE REVIEW

**Comment by jeschkies** [comment]
There are still 65 errors. As a follow up I'd like to factor out the classes.

File: tox.ini [hunk 1]
```diff
@@ -177,16 +177,18 @@ commands=
...
 commands=
   mypy ./test-e2e
   mypy ./packages/cockroach/extra
   mypy ./packages/exhibitor/extra/dcos_zk_backup.py
+  mypy --follow-import=skip ./release/__init__.py
 
 [testenv:collect-integration-tests]
 platform=linux|darwin
-basepython=python3.5
+basepython=python3.6
```
Comment (jeschkies) [inline tox.ini]: According to the styleguide and the README we are on Python 3.6.

**Comment by jeschkies** [comment]
The build fails with
```
ERROR: InterpreterNotFound: python3.6
```
@orsenthil, is there no Python 3.6 on the TeamCity nodes?

**Comment by jeschkies** [comment]
@mesosphere-mergebot label Ready for Review

**Comment by orsenthil** [comment]
Karsten, yeah. We need to upgrade base python in TC nodes.

**Comment by janisz** [comment]
@mesosphere-mergebot status work in progres

**Comment by janisz** [comment]
@mesosphere-mergebot help

**Comment by janisz** [comment]
@mesosphere-mergebot label Work in Progress

**Comment by jeschkies** [comment]
@janisz, I'm dropping this PR. I don't think we will get Python 3.6 on our build machines any time soon.

**Comment by janisz** [comment]
@cgawron1 @DominikDary  Python 3.5 security patches will ends in two months can we get Python 3.6 on our CI?
https://endoflife.date/python

**Pull request closed without merging**